test/transaction.cpp

```diff
@@ -0,0 +1,154 @@
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file transaction.cpp
+ * @author Christoph Jentzsch <cj@ethdev.com>
+ * @date 2014
+ * Transaaction test functions.
+ */
+
+#include <boost/filesystem/operations.hpp>
+#include <boost/test/unit_test.hpp>
+#include "JsonSpiritHeaders.h"
+#include <libdevcore/CommonIO.h>
+#include <libethereum/BlockChain.h>
+#include <libethereum/State.h>
+#include <libethereum/ExtVM.h>
+#include <libethereum/Defaults.h>
+#include <libevm/VM.h>
+#include "TestHelper.h"
+
+using namespace std;
+using namespace json_spirit;
+using namespace dev;
+using namespace dev::eth;
+using namespace dev::eth;
+
+namespace dev {  namespace test {
+
+bytes importByteArray(const std::string& str)
+{
+    bytes data;
+    if (str.find_first_of("0x") == 0)
+            data = fromHex(str.substr(2));
+        else
+            data = fromHex(str);
+    return data;
+}
+
+void doTransactionTests(json_spirit::mValue& v, bool _fillin)
+{
+    processCommandLineOptions();
+
+    for (auto& i: v.get_obj())
+    {
+        cerr << i.first << endl;
+        mObject& o = i.second.get_obj();
+
+        BOOST_REQUIRE(o.count("rlp") > 0);
+        BOOST_REQUIRE(o.count("sender") > 0);
+        BOOST_REQUIRE(o.count("transaction") > 0);
+
+        bytes rlpReaded = importByteArray(o["rlp"].get_str());
+        Address addressReaded = Address(o["sender"].get_str());
+
+        mObject tObj = o["transaction"].get_obj();
+
+        BOOST_REQUIRE(tObj.count("data") > 0);
+        BOOST_REQUIRE(tObj.count("value") > 0);
+        BOOST_REQUIRE(tObj.count("gasPrice") > 0);
+        BOOST_REQUIRE(tObj.count("gasLimit") > 0);
+        BOOST_REQUIRE(tObj.count("nonce")> 0);
+        BOOST_REQUIRE(tObj.count("to") > 0);
+
+        BOOST_REQUIRE(tObj.count("v") > 0);
+        BOOST_REQUIRE(tObj.count("r") > 0);
+        BOOST_REQUIRE(tObj.count("s") > 0);
+
+        Transaction tr = tObj["to"].get_str().empty() ?
+           Transaction(toInt(tObj["value"]), toInt(tObj["gasPrice"]), toInt(tObj["gasLimit"]), importData(tObj), toInt(tObj["nonce"]), Secret(tObj["secretKey"].get_str())) :
+           Transaction(toInt(tObj["value"]), toInt(tObj["gasPrice"]), toInt(tObj["gasLimit"]), Address(tObj["to"].get_str()), importData(tObj), toInt(tObj["nonce"]), Secret(tObj["secretKey"].get_str()));
+
+        //cnote << "Readed RLP: " << toHex(rlpReaded);
+        //cnote << "Sender: " << tr.sender();
+        //cnote << "Transaction RLP: " << toHex(tr.rlp());
+
+        Transaction trFromRLP = Transaction(rlpReaded);
+
+        //Check the fields restored from RLP to original fields
+        if(tr.data() != trFromRLP.data())
+            BOOST_ERROR("Data in given RLP not matching the Transaction data!");
+        if(tr.value() != trFromRLP.value())
```

Comment:
use BOOST_CHECK_MESSAGE(tr.data() == trFromRLP.data(), "Data in given RLP not matching the Transaction data!") instead. (also for all the other cases)